### DISCUSSION TITLE
Question about the command `pyright-langserver`

### DISCUSSION THREAD

**Discussion by acphile** [microsoft/pyright#3515] [General]
Hi, I would like to use Pyright outside the VSCode as a service that after I start the language server, I can send requests mannually which follow the language server protocal and get the response I need.

I guess the command I need is `pyright-langserver` but it seems lack a documentation for how to use it. I try `pyright-langserver --socket=8080` but it just return `Sending log message failed`.

I notice that for python language server like `https://github.com/python-lsp/python-lsp-server`, it can be started by command-line to bind the server to a port and we can send and receive messages from that port by socket. I wonder if `pyright` has similar functionality. 

Thank you very much!

**Comment by erictraut** [comment]
Converting to a discussion topic.

**Comment by erictraut** [comment]
Pyright is leveraged by LSP plugins for several clients including [emacs](https://github.com/emacs-lsp/lsp-pyright) and [vim](https://github.com/fannheyward/coc-pyright). I recommend checking out those as templates for what you want to do.

**Comment by timelikeswind** [comment]
I encountered the same problem. How did you solve it later
